offload/plugins-nextgen/common/src/GlobalHandler.cpp

```diff
@@ -265,7 +273,7 @@ void GPUProfGlobals::dump() const {
 }
 
 Error GPUProfGlobals::write() const {
-  if (!__llvm_write_custom_profile)
+  if (__llvm_write_custom_profile == nullptr)
```

Comment:
```suggestion
  if (!__llvm_write_custom_profile)
```
nit. this is idiomatic and shouldn't have been changed.